packages/core/src/types.ts

```diff
@@ -40,13 +40,21 @@ export interface IValidator<Resource, Input, Config, Output> {
   validateOutput?: (opts: { resource: Resource; output?: Output; config?: Config }) => IPrismDiagnostic[];
 }
 
-export interface IPrismComponents<Resource, Input, Output, Config extends IPrismConfig> {
+type MockerOrForwarder<Resource, Input, Output, Config extends IPrismConfig> =
+  | {
+      forwarder?: IForwarder<Resource, Input, Config, Output>;
+      mocker: IMocker<Resource, Input, Config, Reader<Logger, Either<Error, Output>>>;
+    }
+  | {
+      forwarder: IForwarder<Resource, Input, Config, Output>;
+      mocker?: IMocker<Resource, Input, Config, Reader<Logger, Either<Error, Output>>>;
+    };
+
+export type IPrismComponents<Resource, Input, Output, Config extends IPrismConfig> = {
   router: IRouter<Resource, Input, Config>;
-  forwarder: IForwarder<Resource, Input, Config, Output>;
-  mocker: IMocker<Resource, Input, Config, Reader<Logger, Either<Error, Output>>>;
   validator: IValidator<Resource, Input, Config, Output>;
   logger: Logger;
-}
+} & MockerOrForwarder<Resource, Input, Output, Config>;
```

Comment:
I don’t want us adding anything that says “forwarder” if we can avoid it. We should be talking about “http client” and “mocks” but forwarder needs to be chipped away at not increased.